pybb/models.py

```diff
@@ -463,3 +475,26 @@ class Meta:
 
     def __str__(self):
         return '%s - %s' % (self.poll_answer.topic, self.user)
+
+
+def create_or_check_slug(instance, model, **extra_filters):
+    """
+    returns a unique slug
+
+    :param instance : target instance
+    :param model: needed as instance._meta.model is available since django 1.6
+    :param extra_filters: filters needed for Forum and Topic for their unique_together field
+    """
+    if not instance.slug:
+        instance.slug = slugify(instance.name)
+    slug = instance.slug
```

Comment:
We should rewrite this line to `slug = instance.slug or slugify(instance.name)` and remove two lines above.
I don't like the idea to assign slug and return different slug in one function